networkx/algorithms/isomorphism/isomorphvf2.py

```diff
@@ -846,6 +923,102 @@ def syntactic_feasibility(self, G1_node, G2_node):
         # Otherwise, this node pair is syntactically feasible!
         return True
 
+    def subgraph_is_isomorphic(self):
+        """Returns `True` if a subgraph of ``G1`` is isomorphic to ``G2``.
+
+        Examples
+        --------
+        When creating the `DiGraphMatcher`, the order of the arguments is important
+
+        >>> G = nx.DiGraph([(1, 2), (2, 1), (2, 3), (3, 2)])
+        >>> H = nx.DiGraph(nx.path_graph(5))
+
+        Check whether a subgraph of G is isomorphic to H:
+
+        >>> isomatcher = nx.isomorphism.DiGraphMatcher(G, H)
+        >>> isomatcher.subgraph_is_isomorphic()
+        False
+
+        Check whether as subgraph of H is isomorphic to G:
+
+        >>> isomatcher = nx.isomorphism.DiGraphMatcher(H, G)
+        >>> isomatcher.subgraph_is_isomorphic()
+        True
+        """
+        return super().subgraph_is_isomorphic()
+
+    def subgraph_is_monomorphic(self):
+        """Returns `True` if a subgraph of ``G1`` is monomorphic to ``G2``.
+
+        Examples
+        --------
+        When creating the `DiGraphMatcher`, the order of the arguments is important.
+
+        >>> G = nx.DiGraph([(0, 1), (2, 1), (3, 2)])
+        >>> H = nx.DiGraph([(0, 1), (1, 2), (2, 3), (3, 2)])
+
+        Check whether a subgraph of G is monomorphic to H:
+
+        >>> isomatcher = nx.isomorphism.DiGraphMatcher(G, H)
+        >>> isomatcher.subgraph_is_monomorphic()
+        False
+
+        Check whether as subgraph of H is isomorphic to G:
+
+        >>> isomatcher = nx.isomorphism.DiGraphMatcher(H, G)
+        >>> isomatcher.subgraph_is_monomorphic()
+        True
+        """
+        return super().subgraph_is_monomorphic()
+
+    def subgraph_isomorphisms_iter(self):
+        """Generator over isomorphisms between a subgraph of ``G1`` and ``G2``.
+
+        Examples
+        --------
+        When creating the `DiGraphMatcher`, the order of the arguments is important
+
+        >>> G = nx.DiGraph([(1, 2), (2, 1), (2, 3), (3, 2)])
+        >>> H = nx.DiGraph(nx.path_graph(5))
```

Comment:
The nodes here make it hard to see if a mapping is from H to G or from G to H.
Could we use `G = nx.DiGraph([("A", "B"), ("B", "C"), ("B", "A"), ("C", "B")])`?
Then the result is something like: `{0: 'A', 1: 'B', 2: 'C'}` 

Another choice could be: `G = nx.DiGraph([(101, 102), (102, 101), (102, 103), (103, 102)])`

Also, would it be simpler to make the graphs directed paths:
```python
G = nx.DiGraph([("A", "B"), ("B", "C")])
H = nx.DiGraph(nx.path_graph(5).edges)
```
But maybe that isn't simpler... it removes one direction from each node-pair. But the constructor syntax is longer.